wolfssl/ssl.h

```diff
@@ -2309,7 +2309,7 @@ WOLFSSL_API int wolfSSL_set_SessionTicket_cb(WOLFSSL*,
 
 #define WOLFSSL_TICKET_NAME_SZ 16
 #define WOLFSSL_TICKET_IV_SZ   16
-#define WOLFSSL_TICKET_MAC_SZ  32
+#define WOLFSSL_TICKET_MAC_SZ  WC_MAX_DIGEST_SIZE
```

Comment:
RFC 5077 section 4 states that the ticket construction presented is a recommendation. They gave it the security analysis and recommend it, but you can do anything. We split that between the CreateTicket() function and the callback. The ticket it supposed to be opaque, and I think that CreateTicket should also see it as opaque. It comes up with the parameters we want to put into the ticket, but the actual encryption/decryption should be handled by the callback. Right now the CreateTicket wants 32 byte MACs and always treats the MAC as 32 bytes. What if you are using Poly1305-CHACHA?

Reply:
I found a use case that reproduces the bug that Coverity described.
1. '$./configure --enable-session-ticket --enable-opensslall --enable-tlsx --enable-debug --enable-psk'
2. Run examples/server/server
3. Set a BP in sha512.c at line 719 
`XMEMCPY(hash, sha512->digest, WC_SHA512_DIGEST_SIZE);`
4. Run examples/client/client

Here you can see 64 bytes being copied into the 32 byte buffer, 'hash'.